vpc/service/internal_apis.go

```diff
@@ -407,3 +409,231 @@ func (vpcService *vpcService) doDetachBranchNetworkInterface(ctx context.Context
 		AssociationID: associationID,
 	}, nil
 }
+
+func (vpcService *vpcService) ResetSecurityGroup(ctx context.Context, request *vpcapi.ResetSecurityGroupRequest) (*vpcapi.ResetSecurityGroupResponse, error) {
+	var resetSgTx *sql.Tx
+	var err error
+	var count int
+	var defaultSecurityGroupID, region, account string
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	ctx, span := trace.StartSpan(ctx, "ResetSecurityGroup")
+	defer span.End()
+
+	resetSgTx, err = beginSerializableTx(ctx, vpcService.db)
+	if err != nil {
+		err = errors.Wrap(err, "Unable to begin serializable transaction")
+		tracehelpers.SetStatus(err, span)
+		return &vpcapi.ResetSecurityGroupResponse{}, err
+	}
+	defer func(tx *sql.Tx) {
+		_ = tx.Rollback()
+	}(resetSgTx)
+
+	sgToDelete := request.GetSgId()
+
+	// First get the default security group that we will replace the Reset SG with
+	logger.G(ctx).WithField("resetSg", sgToDelete).Debug("Get Default security group ID of the VPC to use in place of the SG")
+	row := resetSgTx.QueryRowContext(ctx, `
+SELECT region, account, group_id FROM security_groups WHERE  group_name='default' and account IN (SELECT account FROM security_groups WHERE group_id = $1) limit 1
```

Comment:
Again, I don't understand why this can't be:

```sql
SELECT region, account, group_id FROM security_groups WHERE group_name='default' and account = (SELECT account FROM security_groups WHERE group_id = $1 LIMIT 1)
```

Reply:
Changed.